Sources/XCTest/Private/XCTestCase.TearDownBlocksState.swift

```diff
@@ -12,29 +12,42 @@ extension XCTestCase {
     /// Supports async and sync throwing methods.
     final class TeardownBlocksState {
 
+        #if DISABLE_XCTWAITER
+        typealias TeardownBlock = @Sendable () async throws -> Void
+        #else
+        typealias TeardownBlock = () throws -> Void
+        #endif
+
         private var wasFinalized = false
-        private var blocks: [() throws -> Void] = []
+        private var blocks: [TeardownBlock] = []
 
         // We don't want to overload append(_:) below because of how Swift will implicitly promote sync closures to async closures,
         // which can unexpectedly change their semantics in difficult to track down ways.
         //
         // Because of this, we chose the unusual decision to forgo overloading (which is a super sweet language feature <3) to prevent this issue from surprising any contributors to corelibs-xctest
         @available(macOS 12.0, *)
         func appendAsync(_ block: @Sendable @escaping () async throws -> Void) {
+            #if DISABLE_XCTWAITER
+            XCTestCase.subsystemQueue.sync {
+                precondition(wasFinalized == false, "API violation -- attempting to add a teardown block after teardown blocks have been dequeued")
+                blocks.append(block)
+            }
+            #else
             self.append {
                 try awaitUsingExpectation { try await block() }
             }
+            #endif
         }
 
         func append(_ block: @escaping () throws -> Void) {
-            XCTWaiter.subsystemQueue.sync {
+            XCTestCase.subsystemQueue.sync {
                 precondition(wasFinalized == false, "API violation -- attempting to add a teardown block after teardown blocks have been dequeued")
                 blocks.append(block)
```

Comment:
In `DISABLE_XCTWAITER` mode, this is implicitly converting the synchronous `block` to an `async` block, right? I think we also need to wrap that in an `@MainActor` closure to preserve the expected execution context on it?

Reply:
Ah, nice catch!